UndertaleModTool/Converters/UndertaleCachedImageLoader.cs

```diff
@@ -487,53 +478,63 @@ public ImageSource CreateLayerSource(in Layer.LayerTilesData tilesData, in Under
                             continue;
                         }
 
-                        Bitmap resBMP = (Bitmap)TileCache[new(tilesBG.Texture.Name.Content, realID)].Clone();
+                        SKBitmap srcBMP = TileCache[new(tilesBG.Texture.Name.Content, realID)];
+                        tileSurf.Canvas.ResetMatrix();
 
                         switch (id >> 28)
                         {
                             case 1:
-                                resBMP.RotateFlip(RotateFlipType.RotateNoneFlipX);
+                                // Flip X
+                                tileSurf.Canvas.Scale(-1, 1, srcBMP.Width / 2, 0);
                                 break;
                             case 2:
-                                resBMP.RotateFlip(RotateFlipType.RotateNoneFlipY);
+                                // Flip Y
+                                tileSurf.Canvas.Scale(1, -1, 0, srcBMP.Height / 2);
                                 break;
                             case 3:
-                                resBMP.RotateFlip(RotateFlipType.RotateNoneFlipXY);
+                                // Flip X and Y
+                                tileSurf.Canvas.Scale(-1, -1, srcBMP.Width / 2, srcBMP.Height / 2);
                                 break;
                             case 4:
-                                resBMP.RotateFlip(RotateFlipType.Rotate90FlipNone);
+                                tileSurf.Canvas.RotateDegrees(90);
                                 break;
                             case 5:
-                                resBMP.RotateFlip(RotateFlipType.Rotate270FlipY);
+                                tileSurf.Canvas.RotateDegrees(270);
+                                tileSurf.Canvas.Scale(1, -1, 0, srcBMP.Height / 2);
                                 break;
                             case 6:
-                                resBMP.RotateFlip(RotateFlipType.Rotate90FlipY);
+                                tileSurf.Canvas.RotateDegrees(90);
+                                tileSurf.Canvas.Scale(1, -1, 0, srcBMP.Height / 2);
                                 break;
                             case 7:
-                                resBMP.RotateFlip(RotateFlipType.Rotate270FlipNone);
+                                tileSurf.Canvas.RotateDegrees(270);
```

Comment:
would appreciate it, if these also had accompanying comments

Reply:
Okay, and what I should write?
`// Rotate 90 degrees clockwise` and so on?

Reply:
I just added them.